### ISSUE TITLE
rubocop --auto-correct can corrupt MiniTests specs

### ISSUE THREAD

**Issue by MrLeebo** [bbatsov/rubocop#5547]
I just started using Rubocop today (v0.52.1) but there seems to be a bug in some rule where it is inserting line breaks/closing braces inappropriately, corrupting the test file.

Out of all my tests, it only seems to have corrupted 8 files, seems related to a multiline `let` call inside of a `describe` block.

--------

## Expected behavior

Tests that were valid and passing before `rubocop --auto-correct` should be valid and passing after.

## Actual behavior

<img width="311" alt="screen shot 2018-02-05 at 11 24 56 am" src="https://user-images.githubusercontent.com/2754163/35819538-9fe4b1b8-0a68-11e8-9a87-f26bcf0563c0.png">

## Steps to reproduce the problem

I deleted lines from the test and re-ran rubocop until I narrowed down what appears to be the culprit.

```rb
require 'test_helper'

class RubocopTest < ActiveJob::TestCase
  describe "with data" do
    let(:data) {[
      {
        'id' => 345
      }
    ]}
  end
end
```

becomes

```rb
require 'test_helper'

class RubocopTest 
}    stCase
  describe 'with data' do
    let(:data) {
      [
      {
        'id' => 345
      }
    ]}
  end
end
```

If it matters, this is my .rubocop.yml:

```yml
Metrics/LineLength:
  Max: 120

Layout/MultilineMethodCallIndentation:
  EnforcedStyle: indented

Layout/AlignParameters:
  EnforcedStyle: with_fixed_indentation

Layout/IndentHeredoc:
  EnforcedStyle: active_support
```

## RuboCop version

```
0.52.1 (using Parser 2.4.0.2, running on ruby 2.2.0 x86_64-darwin15)
```

**Comment by MrLeebo** [comment]
As a workaround, the `let` can be rewritten as 

```rb
let(:data) do
  [
    { 'id' => 345 }
  ]
end
``` 
or collapsed into a one-liner and it does not have a problem.

**Comment by rrosenblum** [comment]
I have narrowed this down to an issue with `Layout/BlockEndNewline`. I'll try to find a fix for this.
